### ISSUE TITLE
User feedback + Design Directions

### ISSUE THREAD

**Issue by rasagy** [mapbox/osmcha-frontend#2]
Last month, we did a short user feedback session with the data team that have been actively using OSMCha for the daily validation work. Using this ticket to document what was discussed and planning next steps.

Here are the key pain points that were highlighted in the conversation:

### BBox + Filter

- **Favorite filters**: No easy way to keep track of filter combinations that one might use regularly
- **Notifications**: No way to know if there are new changesets for a particular filter — especially useful with one/multiple bbox set as _my area_ and get notifications for new results
- **Logic operators**: Can’t exclude certain values/filters using a NOT (!) operator
- Tag-based filters: No way to extensively filter for a particular tag, like looking for all changesets that add fire hydrant(s)

### List of Changesets

- **Auto refresh list**: Once some changesets are checked, the list doesn’t auto-refresh, so you need to refresh everytime — which is slow if there are several changesets in the result
- **Pagination controls**: You can’t control how many changesets show on each page (set to 15) or jump pages quickly (without editing url) — which makes some low signal queries hard to parse through
- **User details**: Apart from user name and changeset comment, showing more user specific details (like user ratings from HDYC, edit count etc.) can help identify potentially harmful edits quicker

### Individual changeset page

- **Multiple tabs**: Due to the slow speed of changeset map (or loading errors), folks in data team tend to open multiple changesets in tabs (~3) in one go so the others load while they examine one.
- **Discussions**: Not seeing changeset comments in this view means they might spend a lot of time on a changeset that might already have been flagged by the community

### Changeset details & actions

- **User details**: Generally head to OSM to see `join date`, `description`, `edit count`, `diary post count`, `active blocks + history`
- **View changeset in OSM**: This helps see any major visual breakage on a map quickly — the OSM map style shows more detail than Mapbox Streets/Dark style
- **View feature list in OSM**: In some cases, it helps to see _all_ the features, not just the suspicious ones
- **Opening in JOSM**: For complex changesets, opening directly in JOSM doesn’t work (slow/too large bbox) — instead folks in data team tend to zoom into particular part of changeset in OSM or select a feature from the list and then load that bbox or feature in JOSM
- **Actual reason vs possible reason**: No way to specify what was actually wrong in the changeset (for analytics or machine learning)
- **Tracking status**: No way to track if something that was marked harmful was fixed (reverted), or still not reverted, or get notifications if there’s an activity on the changeset (this would be a workflow overhead though)
- **Tracking action**: No way to specify what was done: was a harmful changeset reverted completely, fixed (partially), comment was left on changeset or something else
- **Leaving a comment**: No way to quickly leave a comment, so tend to go to OSM comments for this

### Mark as harmful/valid

- **Actions**: They generally don’t mark all valid changesets, only the harmful ones

cc @batpad @geohacker @ajithranka @kepta @willemarcel

**Comment by rasagy** [comment]
### Next actions:
- [ ] Document common workflows (Why do most users come to OSMCha for? How do they use OSMCha?)
- [ ] Sketch out ideas for the overall flow
- [ ] Sketch out ideas for specific parts of the flow — homepage (list of changesets), filter section, changeset details.

**Comment by ajithranka** [comment]
List of UI components based on current OSMCHA.

- Header
  - Changesets page / Features page toggle
  - Link to Stats page
  - Link to About page
  - Login with OpenStreetMap
- Changesets page
  - Changeset filters
  - Changesets list view with pagination
  - Changeset master view
    - Changeset description with toolbar
    - Changeset filters
    - Changeset map
- Features page
  - Feature filters
  - Features list view with pagination
  - Feature master view
    - Feature description with toolbar
    - Changeset map with the feature highlighted (?)
- Stats page
- About page
- User profile page

**Comment by rasagy** [comment]
Late last week, I did short contextual inquiries with 3 folks from the Mapbox Data Team. Here are the [raw notes](https://gist.github.com/rasagy/38bafd7f32920f6b0119b212da29aae5), with interesting pointers on homepage, which filters are used more often, what’s important in the changeset list, and workflow patterns. Will be referring to these in detail when we start designing specific flows.

**Comment by rasagy** [comment]
Some more feedback in the wild (OSM-talk, Diary post comments) on OSMCha or the needs of such a tool is in [this gist](https://gist.github.com/rasagy/1f306599b3a6d10cd026870a7e147b0a), will keep updating as I come across more such feedback. cc @planemad @maning

**Comment by planemad** [comment]
The goto changeset review tool for the community are:
- http://resultmaps.neis-one.org/osm-suspicious#0/10/8 to find suspicous changesets which has an easy to use UI for filtering changesets by new users
- http://resultmaps.neis-one.org/newestosm#1/15/10 to find new contributors by country
- http://resultmaps.neis-one.org/osm-discussions#1/28/-1 to follow the latest discussions for actionable issues that may require community participation
- http://simon04.dev.openstreetmap.org/whodidit/ to find changes in an area
- https://tyrasd.github.io/osm-qa-feeds/ to subscribe to RSS changes to an area

We've already seen how adding links to existing tools with can quickly increase the adoption of our products https://github.com/osmlab/changeset-map/issues/144 and must definitely see how we can integrate other community triaging tools within osmcha to make it the goto tool for any kind of changeset review.

**Comment by rasagy** [comment]
Few more tools that I noticed in the discussions for comparison/competitive analysis:

- [Keep Right at](https://www.keepright.at/report_map.php?zoom=17&lat=53.39765&lon=-1.51613&layers=B0T&ch=0%252C191%252C192%252C193%252C194%252C195%252C196%252C197%252C198%252C201%252C202%252C203%252C204%252C205%252C206%252C207%252C208%252C210&show_ign=1&show_tmpign=1): List of common errors detected in an area
- [New OSM Contributors in Belgium](https://welcome.osm.be/index.php): Used to welcome new contributors, nice summary of the change
- [Changeset Viewer that shows all bbox on a map](https://osm.expandable.dk/): Gives a more detailed count of added/modiefied/deleted as well as tag-based and type-based.
- [Shadowed Edits](https://mapbox.github.io/osm-analysis-collab/shadowed-edits?layer=0&yearIdx=9&minEdits=1&minBuildings=0&minAmenities=0&minRoads=0&minTotalEdits=0&minUsers=1&minYearEdits=0&query=%22undefined%22&#8/25.089/46.751): Tiles with heatmap like coloring (based on edit count), year scroller, and some useful filters
- Maps.me Edits: http://mmwatch.osmz.ru/

> must definitely see how we can integrate other community triaging tools within osmcha to make it the goto tool for any kind of changeset review.
@planemad We include the changeset map in the detail view, but would be interesting to know which other tools might be useful in a usual OSMCha workflow. Will keep an eye out for this.

**Comment by willemarcel** [comment]
I would like to have in the changeset detail page the links both to the user page in OSM and to a changeset list view filtering that user edits.

In the feature detail page, I would like to have the links to the feature page and to the feature history page (both in OSM website).

**Comment by kepta** [comment]
Closing this ticket as the discussion has moved on to #17 
cc @rasagy
